concourse/pipelines/templates/build_pipeline-tpl.yml

```
@@ -1,1842 +1,336 @@
-        GOOGLE_PROJECT_ID: ((ud/pxf/common/kerberos-google-project-id))
-        GOOGLE_ZONE: ((ud/pxf/common/kerberos-google-zone))
-        HADOOP_USER: gpuser
-        IMAGE_VERSION: ((dataproc-image-version))
-        INITIALIZATION_SCRIPT: gs://data-gpdb-ud-kerberos-scripts/scripts/initialization-for-kerberos.sh
-        INSTANCE_TAGS: bosh-network,data-gpdb-ud-access
-        KERBEROS: ((kerberos-enabled))
-        KEY: dataproc-kerberos-key
-        KEYRING: dataproc-kerberos
-        ccp_reap_minutes: 120
-        NO_ADDRESS: false
-        PROXY_USER: gpuser
-        SECRETS_BUCKET: ((ud/pxf/secrets/kerberos-pxf-secrets-bucket-name))
-  - task: setup_pxf
-    input_mapping:
-      terraform: terraform_gpdb
-    file: pxf_src/concourse/tasks/install_pxf_on_ccp.yml
-    image: ccp-7-image
-    params:
-      IMPERSONATION: false
-      INSTALL_GPHDFS: false
-      GP_VER: [[gp_ver]]
-      KERBEROS: ((kerberos-enabled))
-      PXF_JVM_OPTS: ((pxf-jvm-opts))
-  - task: test_pxf-gp[[gp_ver]]-hdp2-secure-multi-no-impers_on_rhel7
-    image: gpdb[[gp_ver]]-pxf-dev-centos7-image
-    file: pxf_src/concourse/tasks/test_pxf_on_ccp.yml
-    attempts: 2
-    params:
-      ACCESS_KEY_ID: ((tf-machine-access-key-id))
-      SECRET_ACCESS_KEY: ((tf-machine-secret-access-key))
-      HIVE_VERSION: 2
-      IMPERSONATION: false
-      KERBEROS: ((kerberos-enabled))
-      GOOGLE_PROJECT_ID: ((ud/pxf/common/google-project-id))
-      GP_VER: [[gp_ver]]
-      GROUP: security,multiClusterSecurity
-      PXF_JVM_OPTS: ((pxf-jvm-opts))
-  - task: test_pxf_cli
-    on_success:
-      in_parallel:
-        steps:
-        - *destroy_dataproc_1
-        - *destroy_dataproc_2
-        - *destroy_gpdb_cluster
-    image: gpdb[[gp_ver]]-pxf-dev-centos7-image
-    file: pxf_src/concourse/tasks/test_pxf_cli.yml
-{% set gp_ver = None %}
-
-## ---------- Multi-node test for GP7 ----------
-{% set gp_ver = 7 %}
-{% set passed = '[testing_gate_for_pxf-gp]' %}
-{% include 'jobs/multinode-tpl.yml'%}
-
-## ---------- GP7 PXF CLI tests ----------
-{% include 'jobs/cli-tpl.yml' %}
-{% set passed = None %}
-{% set gp_ver = None %}
+{% call(x) macros.for_each_config(build_test_pxf_combinations) %}
+    {% if x.test_multi is sameas true %}
+        {% do x.update({'use_fdw': 'false'}) %}
+        {% do x.update({'use_impers': 'true'}) %}
+        {% do x.update({'target': default_distro}) %}
+        {% do x.update({'distro': default_distro}) %}
+
+        {# include upgrade task when running on centos7 only #}
+        {% set with_upgrade = 'true' if x.test_platform == 'centos7' else 'false' %}
+        {% do x.update({'with_upgrade': with_upgrade}) %}
+
+        {% do x.update({'passed': '[testing-gate-for-pxf-gp]'}) %}
+        {% include 'jobs/test-multinode-tpl.yml' %}
+        [[macros.add_passing_job(x, job_names, 'extended')]]
+    {% endif %}
+    {# multinode no impersonation job for now will only be for GP6 and Centos7 #}
```

```suggestion
    {# multinode no impersonation job for now will only be for GP6 with Centos7 #}
```
Maybe this will be much more clear? When I first read `GP6 and Centos7` I was thinking about everything related to GP6 and everything related to centos 7, but it is GP6 + centos 7 only